apps/prairielearn/src/lib/editors.ts

```diff
@@ -1172,34 +1173,131 @@ export class QuestionAddEditor extends Editor {
       const { shortName, longName } = getUniqueNames({
         shortNames: oldNamesShort,
         longNames: oldNamesLong,
+        shortName: this.qid,
+        longName: this.title,
       });
 
       return { qid: shortName, title: longName };
     });
 
-    const questionPath = path.join(questionsPath, qid);
+    const newQuestionPath = path.join(questionsPath, qid);
 
-    const fromPath = path.join(EXAMPLE_COURSE_PATH, 'questions', 'demo', 'calculation');
-    const toPath = questionPath;
+    // Ensure that the question folder path is fully contained in the questions directory of the course
+    if (!contains(questionsPath, newQuestionPath)) {
+      throw new AugmentedError('Invalid folder path', {
+        info: html`
+          <p>The path of the question folder to add</p>
+          <div class="container">
+            <pre class="bg-dark text-white rounded p-2">${newQuestionPath}</pre>
+          </div>
+          <p>must be inside the root directory</p>
+          <div class="container">
+            <pre class="bg-dark text-white rounded p-2">${questionsPath}</pre>
+          </div>
+        `,
+      });
+    }
 
-    debug(`Copy template\n from ${fromPath}\n to ${toPath}`);
-    await fs.copy(fromPath, toPath, { overwrite: false, errorOnExist: true });
+    if (this.template_qid) {
+      const exampleCourseQuestionsPath = path.join(EXAMPLE_COURSE_PATH, 'questions');
+      const fromPath = path.join(exampleCourseQuestionsPath, this.template_qid);
+
+      // Ensure that the template_qid folder path is fully contained in the example course questions directory
+      if (!contains(exampleCourseQuestionsPath, fromPath)) {
+        throw new AugmentedError('Invalid folder path', {
+          info: html`
+            <p>The path of the template question folder</p>
+            <div class="container">
+              <pre class="bg-dark text-white rounded p-2">${fromPath}</pre>
+            </div>
+            <p>must be inside the root directory</p>
+            <div class="container">
+              <pre class="bg-dark text-white rounded p-2">${exampleCourseQuestionsPath}</pre>
+            </div>
+          `,
+        });
+      }
+
+      const toPath = newQuestionPath;
```

Comment:
Nit: we can probably just use `newQuestionPath` directly instead of aliasing it to `toPath`.